### PR TITLE
chore: release plans flow in flag environments

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-2816/add-release-plan-to-feature-flag-from-release-template
https://linear.app/unleash/issue/2-2818/list-release-plan-with-milestones-in-feature-flag-environment-section
https://linear.app/unleash/issue/2-2819/removing-release-plan-from-feature

Implements the release plan flow in the feature flag environment.

You can now manage release plans in a feature flag environment by adding or removing them, as well as start milestones.

https://github.com/user-attachments/assets/24db9db4-7c3a-463e-b48a-611358f2b212

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlan.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                    <Delete />
+                </PermissionIconButton>
+            </StyledHeader>
+            <StyledBody>
+                {milestones.map((milestone, index) => (
+                    <div key={milestone.id}>
+                        <ReleasePlanMilestone
+                            milestone={milestone}
+                            status={
+                                index === activeIndex
```
Comment (daveleek) [inline frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlan.tsx:170]: Don't we have activeMilestoneId we could use here?

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx [hunk 2]
```diff
@@ -210,74 +230,97 @@ const EnvironmentAccordionBody = ({
...
+                                    releasePlans.length > 0 &&
+                                    strategies.length > 0
+                                }
+                                show={
+                                    <SectionSeparator>
+                                        <StyledBadge>OR</StyledBadge>
+                                    </SectionSeparator>
+                                }
+                            />
+                            <ConditionallyRender
```
Comment (daveleek) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx:250]: Same question around these being moved away from and consistency as further down

**Review by chriswk — APPROVED** [review]
Except for David's question about activeMilestoneId this looks good to me.

File: frontend/src/component/feature/FeatureView/FeatureOverview/NewFeatureOverviewEnvironment/FeatureOverviewEnvironmentBody.tsx [hunk 3]
```diff
@@ -224,78 +245,105 @@ export const FeatureOverviewEnvironmentBody = ({
...
-                            }
-                            show={
-                                <>
-                                    {strategiesToDisplay.map(
-                                        (strategy, index) => (
+                        <>
+                            {releasePlans.map((plan) => (
+                                <ReleasePlan key={plan.id} plan={plan} />
+                            ))}
+                            <ConditionallyRender
```
[the remark this answers is not in the document]
Comment (nunogois) [reply frontend/src/component/feature/FeatureView/FeatureOverview/NewFeatureOverviewEnvironment/FeatureOverviewEnvironmentBody.tsx:257]: Are we? I'm not 100% sure. But like you said this is at least consistent with the rest of the code in the file, so we can return to this later.

File: frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlan.tsx (hunk 1 above)
Comment (nunogois) [reply frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlan.tsx]: We're already [using it](https://github.com/Unleash/unleash/pull/8843/files#diff-498111a84758346c8dba958921663db58596ddb5fdc5082da8b88f7173ea26d7R136-R138) here, even if it's indirectly, so in the end it shouldn't matter much. However I'm on board if you think this is more readable: https://github.com/Unleash/unleash/pull/8843/commits/f42b7883db3d8aaa725cc64c40ae06721c0cd043

**Review by daveleek — APPROVED** [review]
Great!

**Pull request merged**